src/sinks/clickhouse.rs

```diff
@@ -116,15 +113,13 @@ impl HttpSink for ClickhouseConfig {
 
         let uri = encode_uri(&self.host, database, &self.table).expect("Unable to encode uri");
 
-        let builder = Request::builder()
+        let mut builder = Request::builder()
             .method(Method::POST)
             .uri(uri.clone())
             .header("Content-Type", "application/x-ndjson");
 
-        let builder = if let Compression::Gzip = self.compression.unwrap_or(Compression::Gzip) {
-            builder.header("Content-Encoding", "gzip")
-        } else {
-            builder
+        if self.compression == Compression::Gzip {
+            builder = builder.header("Content-Encoding", "gzip")
```

Comment:
Suggestion: Add a method to `impl Compression` that returns the right (optional) content encoding string.